runtime/src/main/java/org/corfudb/runtime/view/replication/IReplicationProtocol.java

```diff
@@ -35,6 +35,17 @@
      */
     void write(Layout layout, ILogData data) throws OverwriteException;
 
+    /** Write forward pointer to the log for a given address
+     *
+     * This function is non-blocking
+     *
+     * @param layout
+     * @param streamId
+     * @param address
+     * @param next
+     */
+    void writeForwardpointer(Layout layout, UUID streamId, long address, long next); /* throws in case of mismatched pointer? */
```

Comment:
It raises a question:  do we need to catch an error here or assume the pointers always match?